mewarpx/mewarpx/diags_store/particle_diagnostic.py

```diff
@@ -112,38 +112,42 @@ def check_for_end_of_sim(self):
-                    level=0,
-                    left_edge=yt_data.domain_left_edge,
-                    dims=yt_data.domain_dimensions
-                )
-                for species_name in self.plot_species:
-                    for param in self.plot_data_list:
-                        field = (species_name, param)
-                        if field not in yt_data.field_list:
-                            warnings.warn(f'{field} '
-                                           'not found in yt_data field list')
-                        else:
-                            raw_data = grid_data[(species_name, param)]
-                            np_data = raw_data.to_ndarray()
-                            fig, ax = plt.subplots(figsize=(14, 14))
-                            ax.hist(np_data, bins=500, histtype='stepfilled',
-                                    alpha=0.85)
-                            ax.set_xlabel(param)
-                            ax.set_ylabel("Counts")
-                            ax.set_title(species_name)
-                            out_path = os.path.join(
-                                self.write_dir,
-                                f'{species_name}_{param}_{step}.jpg',
-                            )
-                            plt.savefig(out_path)
+                if not "old" in datafolder:
+
+                    print('Reading', datafolder, '\n')
+                    step = datafolder.replace(self.write_dir + "/" + self.name, "")
+                    yt_data = yt.load(datafolder)
+                    grid_data = yt_data.covering_grid(
+                        level=0,
+                        left_edge=yt_data.domain_left_edge,
+                        dims=yt_data.domain_dimensions
+                    )
+                    for species_name in self.plot_species:
+                        for param in self.plot_data_list:
+                            field = (species_name, param)
+                            print(f"This is field list: {yt_data.field_list}")
+                            if field not in yt_data.field_list:
+                                warnings.warn(f'{field} '
+                                            'not found in yt_data field list')
+                            else:
+                                raw_data = grid_data[(species_name, param)]
+                                np_data = raw_data.to_ndarray()
+                                fig, ax = plt.subplots(figsize=(14, 14))
+                                ax.hist(np_data, bins=500, histtype='stepfilled',
+                                        alpha=0.85)
+                                ax.set_xlabel(param)
+                                ax.set_ylabel("Counts")
+                                ax.set_title(species_name)
+                                out_path = os.path.join(
+                                    self.write_dir,
+                                    f'{species_name}_{param}_{step}.jpg',
```

Comment:
Can we switch this to save the image as a `.png` file so that we have consistency with our other diagnostics.